src/Network/Minio/Data.hs

```diff
@@ -182,6 +182,47 @@ type Region = Text
 -- APIs.
 type ETag = Text
 
+-- |
+-- Data type represents various options specified for PutObject call
+data PutObjectOptions = PutObjectOptions {
+    pooContentType :: Maybe Text
+  , pooContentEncoding :: Maybe Text
+  , pooContentDisposition :: Maybe Text
+  , pooCacheControl :: Maybe Text
+  , pooUserMetadata :: [(Text, Text)]
+  , pooNumThreads :: Maybe Word
+    } deriving (Show, Eq)
+
+-- We expose the data type PutObjectOptions (but not the
+-- constructor of the same name), and we expose all the members.
+
+-- Provide a default instance
+instance Default PutObjectOptions where
+    def = PutObjectOptions def def def def [] def
+
+addXAmzMetaPrefix :: ByteString -> ByteString
+addXAmzMetaPrefix s = do
+  if (B.isPrefixOf (encodeUtf8 "x-amz-meta-") s)
+    then s
+    else B.concat [(encodeUtf8 "x-amz-meta-"), s]
```

Comment:
It is not necessary to use `encodeUtf8` as only the literal string `"x-amz-meta-"` is being used. The [OverloadedStrings](https://www.schoolofhaskell.com/school/to-infinity-and-beyond/pick-of-the-week/guide-to-ghc-extensions/basic-syntax-extensions#overloadedstrings) extension takes care of interpreting the literal under the inferred type.